src/LinearMaps.jl

```diff
@@ -20,87 +19,55 @@ Base.eltype(::LinearMap{T}) where {T} = T
 Base.eltype(::Type{L}) where {T,L<:LinearMap{T}} = T
 
 Base.isreal(A::LinearMap) = eltype(A) <: Real
-Base.issymmetric(::LinearMap) = false # default assumptions
-Base.ishermitian(A::LinearMap{<:Real}) = issymmetric(A)
-Base.ishermitian(::LinearMap) = false # default assumptions
-Base.isposdef(::LinearMap) = false # default assumptions
+issymmetric(::LinearMap) = false # default assumptions
+ishermitian(A::LinearMap{<:Real}) = issymmetric(A)
+ishermitian(::LinearMap) = false # default assumptions
+isposdef(::LinearMap) = false # default assumptions
 
 Base.ndims(::LinearMap) = 2
 Base.size(A::LinearMap, n) = (n==1 || n==2 ? size(A)[n] : error("LinearMap objects have only 2 dimensions"))
 Base.length(A::LinearMap) = size(A)[1] * size(A)[2]
 
-# any LinearMap subtype will have to overwrite at least one of the two following methods to avoid running in circles
-*(A::LinearMap, x::AbstractVector) = Base.A_mul_B!(similar(x, promote_type(eltype(A),eltype(x)), size(A,1)), A, x)
-Base.A_mul_B!(y::AbstractVector, A::LinearMap, x::AbstractVector) = begin
-    length(y) == size(A,1) || throw(DimensionMismatch("A_mul_B!"))
-    copy!(y, A*x)
+*(A::LinearMap, x::AbstractVector) = mul!(similar(x, promote_type(eltype(A),eltype(x)), size(A,1)), A, x)
+mul!(y::AbstractVector, A::LinearMap, x::AbstractVector) = begin
+    length(y) == size(A,1) || throw(DimensionMismatch("mul!"))
```

Comment:
Similarly here: `Base.:(+)` (possibly) and `LinearAlgebra.mul!` (preferably). 

Also, I don't know why I choose to write the `mul!` function as
```
function mul!( ) = begin
... (multiple lines)
end
```
instead of just
```
function mul!( )
... multiple lines
end
```